kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3Storage.java

```diff
@@ -101,16 +106,45 @@ public ClientConfiguration newClientConfiguration(S3SinkConnectorConfig config)
     clientConfiguration.withUserAgentPrefix(version);
     if (StringUtils.isNotBlank(config.getString(S3_PROXY_URL_CONFIG))) {
       S3ProxyConfig proxyConfig = new S3ProxyConfig(config);
+      RetryPolicy fullJitterPolicy = newRetryPolicy(config);
       clientConfiguration.withProtocol(proxyConfig.protocol())
           .withProxyHost(proxyConfig.host())
           .withProxyPort(proxyConfig.port())
           .withProxyUsername(proxyConfig.user())
-          .withProxyPassword(proxyConfig.pass());
+          .withProxyPassword(proxyConfig.pass())
+          .withRetryPolicy(fullJitterPolicy);
     }
 
     return clientConfiguration;
   }
 
+
+  /**
+   * Creates a retry policy, based on full jitter backoff strategy
+   * and default retry condition.
+   *
+   * @param config the S3 configuration.
+   * @return retry policy
+   * @see com.amazonaws.retry.PredefinedRetryPolicies.SDKDefaultRetryCondition
+   * @see PredefinedBackoffStrategies.FullJitterBackoffStrategy
+   */
+  public RetryPolicy newRetryPolicy(S3SinkConnectorConfig config) {
+
+    PredefinedBackoffStrategies.FullJitterBackoffStrategy backoffStrategy = new
+        PredefinedBackoffStrategies.FullJitterBackoffStrategy(
+        config.getLong(RETRY_BACKOFF_CONFIG).intValue(),
+        MAX_RETRY_TIME_MS
+    );
+
+    RetryPolicy retryPolicy = new RetryPolicy(
+        PredefinedRetryPolicies.DEFAULT_RETRY_CONDITION,
+        backoffStrategy,
+        conf.getS3PartRetries(),
```

Comment:
Need to describe how the `s3.part.retries` property affects retry delays.

Reply:
Ok, will add explanation and example as in ES.

Reply:
I like the additions. Thanks.